routers/api/v2/users.go

```diff
@@ -443,6 +442,53 @@ func (r *apiV2Router) SetRole(ctx *gin.Context) {
 	ctx.Status(http.StatusNoContent)
 }
 
+// PUT: /api/v2/users/:id/permissions
+// x-www-form-urlencoded
+// Request:  permissions string
+// Headers:  Authorization -> token
+func (r *apiV2Router) SetSpecialPermissions(ctx *gin.Context) {
+	var (
+		err error
+		req struct {
+			Permissions string `form:"permissions"`
```

Comment:
Would `Permissions common.UniformResourceIdentifiers `form:"permissions"`` not work since we're just running `json.Unmarshal` on the string?

Reply:
In order to directly bind to `common.UniformResourceIdentifiers` I'd need to change the request body from `application/x-www-form-encoded` to `application/json`. Then I can using `ctx.BindJSON` which will automatically call the `UnmarshallJSON` function on `common.UniformResourceIdentifiers`.

IMO does make more sense for the API endpoint to accept `application/json` anyway

Reply:
Would that change how the operation is supposed to be used from the user's perspective?

Reply:
I don't believe so, in the client we'd just have to make sure to set the `Content-Type` header correctly

Reply:
Yeah that sounds like a good solution